client/starwhale/core/job/scheduler.py

```diff
@@ -0,0 +1,258 @@
+import os
+import time
+import typing as t
+import threading
+import concurrent.futures
+from abc import abstractmethod
+from pathlib import Path
+from multiprocessing.connection import Connection
+
+from loguru import logger
+
+from starwhale.consts import EvaluationResultKind
+from starwhale.api._impl import wrapper
+from starwhale.utils.config import SWCliConfigMixed
+from starwhale.core.job.model import Step, Task, STATUS
+from starwhale.api._impl.wrapper import (
+    EvaluationQuery,
+    EvaluationMetric,
+    EvaluationResult,
+)
+
+
+class TaskPipe:
+    def __init__(self, idx: str, main_conn: Connection, sub_conn: Connection):
+        self.id = idx
+        self.main_conn = main_conn
+        self.sub_conn = sub_conn
+
+
+def _init_datastore(root_path: str, project: str, eval_id: str) -> wrapper.Evaluation:
+    os.environ["SW_ROOT_PATH"] = root_path
```

Comment:
`SW_ROOT_PATH` is removed.  related pr: https://github.com/star-whale/starwhale/pull/902